pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -399,6 +404,298 @@ async def _map_user_prompt(part: UserPromptPart) -> chat.ChatCompletionUserMessa
         return chat.ChatCompletionUserMessageParam(role='user', content=content)
 
 
+@dataclass(init=False)
+class OpenAIResponsesModel(Model):
+    """A model that uses the OpenAI Responses API."""
+
+    client: AsyncOpenAI = field(repr=False)
+    system_prompt_role: OpenAISystemPromptRole | None = field(default=None)
+
+    _model_name: OpenAIModelName = field(repr=False)
+    _system: str = field(default='openai', repr=False)
+
+    def __init__(
+        self,
+        model_name: OpenAIModelName,
+        *,
+        provider: Literal['openai', 'deepseek', 'azure'] | Provider[AsyncOpenAI] = 'openai',
```

Comment:
only partially related, but should we add `ollama` or `local` here? Do all the local models default to the same port?

Reply:
We should, we just didn't yet. We need to implement their providers.